### PR TITLE
Add Pydantic Models for Sparsify CLIs

### PR DESCRIPTION
**Pull request by rahul-tuli**
This PR adds Pydantic Models for all sparsify CLI(s) namely
 * `sparsify.init`
 * `sparsify.apply`
 * `sparsify.package`
 * `sparsify.login`

The args have been taken from the following:
https://github.com/neuralmagic/internal-docs/blob/main/products/sparsify/clis/drawio-assets/sparsify-clis-Page-1.png

### CODE REVIEW

File: src/sparsify/cli/utils/args.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    )
+
+class APIKey(BaseModel):
+    """
+    API Key specific arguments for sparsify
+    """
+
+    api_key: Optional[str] = Field(
+        default=None, description="The sparsify API key"
+    )
```
Comment (bfineran) [inline src/sparsify/cli/utils/args.py:177]: no new line

File: src/sparsify/cli/utils/args.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        description="The optimization level to use, must be a float b/w [0-1]",
+    )
+    optim_for_hardware: Optional[float] = Field(
+        default=None,
+        description="The optimization level for current hardware, "
+        "must be a float b/w [0-1]",
+    )
+
+
+class InfoArgs(BaseModel):
```
Comment (bfineran) [inline src/sparsify/cli/utils/args.py:141]: let's call `ProjectArgs`

**Review by bfineran — APPROVED** [review]

File: src/sparsify/cli/utils/args.py (hunk 2 above)
Comment (rahul-tuli) [reply src/sparsify/cli/utils/args.py:141]: Discussed offline this is called out as `InfoArgs` in product doc

File: src/sparsify/cli/utils/args.py (hunk 1 above)
Comment (rahul-tuli) [reply src/sparsify/cli/utils/args.py:177]: Added

**Pull request merged**